### DISCUSSION TITLE
docs/menu-permissions/

### DISCUSSION THREAD

**Discussion by giscus** [casbin/casbin#1375] [Docs comments]
# docs/menu-permissions/

Example for Menu Permissions

https://casbin.org/docs/menu-permissions/

**Comment by huxulm** [comment]
hello, can you tell me where was wrong? below is my test code,  but i cant get the result as expected.

```go
func TestMenu(t *testing.T) {
	f := getEnforcer(t, model_2, policy_2)
	PERMS := []string{
		"SystemMenu",        //	✅	❌	❌
		"UserMenu",           //	❌	✅	✅
		"UserSubMenu_allow",  //	❌	✅	✅
		"UserSubSubMenu",     //	❌	✅	✅
		"UserSubMenu_deny",   //	❌	✅	❌
		"AdminMenu",          //	✅	✅	❌
		"AdminSubMenu_allow", //	✅	✅	❌
		"AdminSubMenu_deny",  //	✅	❌	❌
	}
	ans := bufio.NewWriter(os.Stdout)
	for _, obj := range PERMS {
		ans.WriteString(obj + strings.Repeat(" ", 30-len(obj)))
		for _, sub := range []string{"ROLE_ROOT", "ROLE_ADMIN", "ROLE_USER"} {
			ok, err := f.Enforce(sub, obj, "read")
			if ok {
				ans.WriteString("✅	")
			} else {
				ans.WriteString("❌	")
			}
			if err != nil {
				t.Fatal(err)
			}
		}
		ans.WriteString("\n")
	}
	ans.Flush()
}
```
the code output is:
```
go test -v -count=1 -run ^TestMenu ./
=== RUN   TestMenu
SystemMenu                    ❌        ❌      ❌
UserMenu                      ❌        ✅      ❌
UserSubMenu_allow             ❌        ✅      ✅
UserSubSubMenu                ❌        ✅      ✅
UserSubMenu_deny              ❌        ✅      ❌
AdminMenu                     ✅        ✅      ❌
AdminSubMenu_allow            ✅        ✅      ❌
AdminSubMenu_deny             ✅        ❌      ❌
--- PASS: TestMenu (0.00s)
PASS
```

**Reply by hsluoyz** [reply]
Hi @huxulm , the Java test cases have been added: https://github.com/jcasbin/jcasbin-menu-permission/pull/12 , it turns out that that repo has a name typo issue, it should be named as "SystemMenu" instead of "OtherMenu" in Casbin policy. It has been fixed in: https://github.com/jcasbin/jcasbin-menu-permission/commit/9ab6039d9e1ff954a85b1ef9171b9faac9d518a4

And an enforcement mismatch error has also been fixed in docs: https://github.com/casbin/casbin-website-v2/commit/1a1d5f8971b37a924e90c3ca4893935b9a796c4f

![image](https://github.com/casbin/casbin/assets/3787410/f17ba265-d27b-4461-91c1-a610b5b4f590)

Now your Go code should get the same result as latest docs
